crates/metastore/src/client.rs

```diff
@@ -323,6 +323,28 @@ impl StatefulWorker {
                     error!("failed to respond to ping");
                 }
             }
+            ClientRequest::Commit {
+                version,
+                state,
+                response,
+            } => {
+                let state: CatalogState = state.as_ref().clone();
+                let commit_request = tonic::Request::new(CommitRequest {
+                    db_id: self.db_id.into_bytes().to_vec(),
+                    catalog_version: version,
+                    catalog: Some(state.try_into().unwrap()),
+                });
+
+                let result = self.client.commit_catalog(commit_request).await.unwrap();
+                let result = result.into_inner();
+                let state: CatalogState = result.catalog.unwrap().try_into().unwrap();
+                self.set_cached_state(state);
+                let res = Ok(self.cached_state.clone());
+
+                if response.send(res).is_err() {
+                    error!("failed to respond to commit");
```

Comment:
should we report more context here (about which DB/catalog failed, or the error itself?

Reply:
no, this is in par with the other error messages

```rs
error!("failed to respond to ping");

error!("failed to send cached state");
```